pkg/bundle/client_test.go

```diff
@@ -0,0 +1,200 @@
+
+	api "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+)
+
+func TestNewPackageBundleClient(t *testing.T) {
+	t.Parallel()
+
+	t.Run("golden path", func(t *testing.T) {
+		t.Parallel()
+
+		mockClient := newMockClient(nil)
+		bundleClient := NewPackageBundleClient(mockClient)
+
+		assert.NotNil(t, bundleClient)
+	})
+}
+
+func TestBundleClient_IsActive(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	t.Run("golden path returning true", func(t *testing.T) {
+		mockClient := newMockClient(nil)
+		bundleClient := NewPackageBundleClient(mockClient)
+		bundle := givenPackageBundle(api.PackageBundleStateInactive)
+
+		active, err := bundleClient.IsActive(ctx, bundle)
+
+		assert.True(t, active)
+		assert.Nil(t, err)
+	})
+
+	t.Run("error on failed get", func(t *testing.T) {
+		mockClient := newMockClient(fmt.Errorf("failed get"))
+		bundleClient := NewPackageBundleClient(mockClient)
+		bundle := givenPackageBundle(api.PackageBundleStateInactive)
+
+		_, err := bundleClient.IsActive(ctx, bundle)
+
+		assert.NotNil(t, err)
+	})
+
+	t.Run("fails on wrong namespace", func(t *testing.T) {
+		mockClient := newMockClient(nil)
+		bundleClient := NewPackageBundleClient(mockClient)
+		bundle := &api.PackageBundle{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "Wrong-Name",
+			},
+			Status: api.PackageBundleStatus{
+				State: api.PackageBundleStateActive,
+			},
+		}
+
+		active, err := bundleClient.IsActive(ctx, bundle)
+
+		assert.False(t, active)
+		assert.Nil(t, err)
+	})
+
+	t.Run("fails on wrong name", func(t *testing.T) {
+		mockClient := newMockClient(nil)
+		bundleClient := NewPackageBundleClient(mockClient)
+		bundle := &api.PackageBundle{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "non-empty",
+				Namespace: api.PackageNamespace,
+			},
+			Status: api.PackageBundleStatus{
+				State: api.PackageBundleStateActive,
+			},
+		}
+
+		active, err := bundleClient.IsActive(ctx, bundle)
+
+		assert.False(t, active)
+		assert.Nil(t, err)
+	})
+}
+
+func TestBundleClient_GetActiveBundle(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	t.Run("golden path", func(t *testing.T) {
+		mockClient := newMockClient(nil)
+		bundleClient := NewPackageBundleClient(mockClient)
+
+		bundle, err := bundleClient.GetActiveBundle(ctx)
+
+		assert.Equal(t, bundle.Name, "test")
+		assert.Nil(t, err)
+	})
+}
+
+func TestBundleClient_GetActiveBundleNamespacedName(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	t.Run("golden path", func(t *testing.T) {
+		mockClient := newMockClient(nil)
+		bundleClient := NewPackageBundleClient(mockClient)
+
+		namespacedNames, err := bundleClient.GetActiveBundleNamespacedName(ctx)
+
+		assert.Equal(t, api.PackageNamespace, namespacedNames.Namespace)
+		assert.Equal(t, "", namespacedNames.Name)
+		assert.Nil(t, err)
+	})
+}
+
+// Helpers
+type mockClient struct {
+	client.Client
+	err error
+}
+
+func newMockClient(err error) *mockClient {
```

Comment:
Can this be thrown in this https://github.com/aws/eks-anywhere-packages/blob/main/Makefile#L163

Reply:
Addressed in latest commit. Used the mock client interface that was already generated in controllers/mocks